runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java

```diff
@@ -201,7 +201,7 @@ public void stop() {
   private GrpcFnServer<InMemoryJobService> createJobServer() throws IOException {
     InMemoryJobService service = createJobService();
     GrpcFnServer<InMemoryJobService> jobServiceGrpcFnServer;
-    if (Strings.isNullOrEmpty(configuration.host)) {
+    if (configuration.port == 0) {
```

Comment:
Should this be `Strings.isNullOrEmpty(configuration.host) || configuration.port == 0`?

Reply:
This makes the behavior less consistent.
Example scenarios:
When host=null : JobPort Dynamic, ArtifactPort Dynamic
When host=localhost: JobPort 8099, ArtifactPort  8098
When host=localhost, jobPort=0: JobPort dynamic, ArtifactPort  8098
....
Just checking if the JobPort is 0 will remove this confusion

Reply:
Yes, that is better. Do you think we could also respect the host name when the port is 0? As of now, it would get ignored.

Reply:
Yes, I think we should but there is no method as of now to do so. I think we should rethink and refactor GrpcFnServer.

Reply:
Ok, let's do that in a follow-up.